cocos/base/Scheduler.cpp

```diff
@@ -177,39 +151,36 @@ void Scheduler::schedule(const ccSchedulerFunc &callback, void *target, float in
     CCASSERT(target, "Argument target must be non-nullptr");
     CCASSERT(!key.empty(), "key should not be empty!");
 
-    tHashTimerEntry *element = nullptr;
-    HASH_FIND_PTR(_hashForTimers, &target, element);
-
-    if (!element) {
-        element = (tHashTimerEntry *)calloc(sizeof(*element), 1);
+    HashTimerEntry *element = nullptr;
+    auto            iter    = _hashForTimers.find(&target);
+    if (iter == _hashForTimers.end()) {
+        element         = new HashTimerEntry();
         element->target = target;
 
-        HASH_ADD_PTR(_hashForTimers, target, element);
+        _hashForTimers[target] = element;
 
         // Is this the 1st element ? Then set the pause level to all the selectors of this target
         element->paused = paused;
     } else {
         CCASSERT(element->paused == paused, "element's paused should be paused!");
     }
 
-    if (element->timers == nullptr) {
-        element->timers = ccArrayNew(10);
+    if (element->timers.empty()) {
+        element->timers.reserve(INITIAL_TIMER_COUND);
```

Comment:
Can this be omitted, since it is still empty after `reserve`?

Reply:
reserve just make sure to have enough memory, do not have to expand every time. Omit it will not cause error, but here just keep the same logic as before.